event-queue/kafka/src/main/java/com/netflix/conductor/contribs/queue/kafka/KafkaTransactionRecord.java

```diff
@@ -37,22 +38,33 @@ public KafkaTransactionRecord(ConsumerRecord<String, String> record, TracingProv
     this.tracingProvider = tracingProvider;
 
     Headers headers = record.headers();
-    Header header = headers.lastHeader(tracingProvider.getTraceHeader());
-    if (header != null) {
-      this.traceHeader = Optional.of(new String(header.value(), StandardCharsets.UTF_8));
+    Header traceHeader = headers.lastHeader(tracingProvider.getTraceHeader());
+    Header userIdentifier = headers.lastHeader("user-uuid");
+    if (traceHeader != null) {
+      this.traceHeader = Optional.of(new String(traceHeader.value(), StandardCharsets.UTF_8));
+      this.userIdentifier = Optional.of(new String(userIdentifier.value(), StandardCharsets.UTF_8));
```

Comment:
I think we may want to propagate the user uuid anyways, we can use it for the logs as well to easily identify the user.